### ISSUE TITLE
508-defect-1 ❗ Launchblocker [SCREENREADER]: Focus MUST move to notification of context change when new content is added

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#7243]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

When composing search criteria, if the user selects Urgent Care and then Community urgent care providers, an alert box appears for sighted users, but there is no notification provided for assistive technologies. These changes **must** be communicated for assistive technologies. JAWS and NVDA screen readerS moved focus from the Search button to the first search result, with no notification of the alert box.

Additionally, "Click below…" with the blue CTA is problematic phrasing here. Recommend checking with @peggygannon for copywriting expertise. 

Success Criteria 4.1.2 - Name, Role, Value: For all user interface components (including but not limited to: form elements, links and components generated by scripts), the name and role can be programmatically determined; states, properties, and values that can be set by the user can be programmatically set; and **notification of changes to these items is available to user agents, including assistive technologies. (Level A)**

## Screenshots or Trace Logs

![Screen Shot 2020-03-26 at 11 34 21 AM](https://user-images.githubusercontent.com/57469/77665244-c7033180-6f55-11ea-8baf-3ffe1a964940.png)

## Point of Contact

**VFS Point of Contact:** Jennifer

## Steps to Recreate

1. Enter `https://staging.va.gov/find-locations/` in browser
2. Start screenreading device listed in Environment
3. Enter search criteria, select Urgent care under facility type, and Community urgent care providers under service type
4. Verify that the alert box with "Important information" appears, but focus remains on the select element

## Possible Fixes

1. Upon selection of Community urgent care providers, the alert should appear near that selection and focus should move to the Important information heading.
1. The Learn more link in the alert should be changed from a blue button to a link (preferable) or a secondary button.
1. The Search button should be moved to beneath the select elements, in order to accommodate the area needed for this alert, and would align with the design system pattern which has buttons in the lower right, left-aligned with the relevant inputs

![Screen Shot 2020-03-26 at 11 49 04 AM](https://user-images.githubusercontent.com/57469/77668119-6249d600-6f59-11ea-89ec-3021442e331f.png)

## WCAG or Vendor Guidance (optional)

* [W3C - Understanding Success Criterion 3.2.1: On Focus](https://www.w3.org/WAI/WCAG21/Understanding/on-focus.html) 
* [W3C - Name, Role, Value](https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html)

**Comment by mmiddaugh** [comment]
@jenstrickland 
We're struggling with how best to manage the focus on this issue given the current workflow. 

- The alert displays once Community urgent care providers is selected as service type for urgent care. 
- If the user clicks to "Learn more...", a new tab will open and the user is taken to that page before viewing search results. Ideally, the focus should move to the search results before the alert based on the workflow. However, it's difficult to move focus to the search results because the results summary subhead is currently hidden [#6775] due to inaccuracy in the calculation of search results. 
- We could use the "Search" button as a trigger for displaying the alert but the same issue resulting from focus moving to the new page before viewing search results would still exist.

Other considerations:
- Because the alert displays only when _Facility type = Urgent care + Service type = Community urgent care providers_ is selected as service type for urgent care, it doesn't make sense to move the search button below the alert. 
- We plan to do a "mashup" of search results soon, which would mean the search results would include both VA and Community care locations. This would force the alert content to move into individual search results so the current alert will no longer display. 

Given all of this....is there short term solution that makes sense or will it be acceptable to address focus when we adjust the workflow?

**Comment by jenstrickland** [comment]
@mmiddaugh I appreciate the details. This is so complex! 

- There's extra complexity for non-sighted users. In the current design, assistive tech users don't get the alert information. The alert currently doesn't receive any focus. Moving it before the Search button would address that. 

- What about putting the text above, though it's provided to all… Such as a heading, **If you plan to go to a Community Urgent Care provider**, then the text, with the link?

Perhaps that would be confusing for more folks.

**Comment by mmiddaugh** [comment]
Blocking this issue pending design iterations

**Comment by jenstrickland** [comment]
@mmiddaugh While validating a different ticket I noticed that the focus stays on the Search button still, and doesn't move an H2 for the Search Results. Is it possible to have this heading added, and focus on Search button input flow:

1. Loading message
1. H2 for Search Results

I know there are inaccuracies with the number of results returned, but just the heading would be helpful. At this time, when a user enters criteria, then hits Search, nothing happens, so there's no feedback that context changed.

**Comment by mmiddaugh** [comment]
Focus has been added to the AC for [#7555](https://github.com/department-of-veterans-affairs/va.gov-team/issues/7555)

**Comment by jenstrickland** [comment]
Validated.
